lib/plugins/aws/deployFunction/index.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+const BbPromise = require('bluebird');
+const fs = require('fs');
+const SDK = require('../');
+const validate = require('../lib/validate');
+
+// The Package plugin which is used to zip the service
+const Package = require('../../package');
+
+class AwsDeployFunction {
+  constructor(serverless, options) {
+    this.serverless = serverless;
+    this.options = options || {};
+    this.provider = 'aws';
+    this.sdk = new SDK(serverless);
+
+    this.functionName =
```

Comment:
We should put this in a central location at some point to make sure we use the same code here and when actually creating the function